NotificationService/NotificationService.BusinessLibrary/Business/v1/EmailManager.cs

```diff
@@ -129,7 +130,13 @@ public async Task<IList<EmailNotificationItemEntity>> CreateNotificationEntities
                 notificationEntities.Add(notificationEntity);
             }
 
-            await this.emailNotificationRepository.CreateEmailNotificationItemEntities(notificationEntities, applicationName).ConfigureAwait(false);
+            var batchesToCreate = BusinessUtilities.SplitList<EmailNotificationItemEntity>(notificationEntities, ApplicationConstants.BatchSizeToStore).ToList();
```

Comment:
This will be applied to both Cosmos and Storage. I guess this is additional logic on top of batching to Queue, which may confuse more. No need of this. In storagehelper, where we create a batch request to insert,, do that change there.